shapely/ops.py

```diff
@@ -498,39 +498,43 @@ def substring(geom, start_dist, end_dist, normalized=False):
     """
 
     assert(isinstance(geom, LineString))
-    
+
     # Filter out cases in which to return a point
     if start_dist == end_dist:
         return geom.interpolate(start_dist, normalized)
     elif not normalized and start_dist >= geom.length and end_dist >= geom.length:
         return geom.interpolate(geom.length, normalized)
     elif not normalized and -start_dist >= geom.length and -end_dist >= geom.length:
-        return geom.interpolate(0, normalized)                    
+        return geom.interpolate(0, normalized)
     elif normalized and start_dist >= 1 and end_dist >= 1:
-        return geom.interpolate(1, normalized)  
+        return geom.interpolate(1, normalized)
     elif normalized and -start_dist >= 1 and -end_dist >= 1:
         return geom.interpolate(0, normalized)
 
     start_point = geom.interpolate(start_dist, normalized)
     end_point = geom.interpolate(end_dist, normalized)
-    
+
     min_dist = min(start_dist, end_dist)
     max_dist = max(start_dist, end_dist)
     if normalized:
         min_dist *= geom.length
         max_dist *= geom.length
-    
+
     if start_dist < end_dist:
         vertex_list = [(start_point.x, start_point.y)]
     else:
         vertex_list = [(end_point.x, end_point.y)]
+
     coords = list(geom.coords)
-    for p in coords:
-        pd = geom.project(Point(p))
-        if min_dist < pd < max_dist:
-            vertex_list.append(p)
-        elif pd >= max_dist:
+    current_distance = 0
+    for p1, p2 in zip(coords, coords[1:]):
+        if min_dist < current_distance < max_dist:
+            vertex_list.append(p1)
+        elif current_distance >= max_dist:
             break
+
+        current_distance += LineString([p1, p2]).length
```

Comment:
To me it feels overkill to instantiate a LineString to calculate the distance between two points.
```
current_distance = ((p2.x - p1.x) ** 2 + (p2.y - p1.y) ** 2) ** 0.5
```

Reply:
Yes, good point